### PR TITLE
Add document to setup ESLint for React Native apps

### PR DESCRIPTION
**Pull request by ghost**
Added ESLint doc with configuration file images

### CODE REVIEW

File: react-native-eslint.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+React Native comes with a predefined .eslintrc.js file in the root folder of your project, Open `.eslintrc.js` file and modify it like the below configuration.
+
+```js
+module.exports = {
+  env: {
+    browser: true,
+    es2021: true,
+    "react-native/react-native": true,
+  },
+  extends: ["plugin:react/recommended", "airbnb", "airbnb/hooks"],
```
Comment (Abijeet) [inline react-native-eslint.md]: Are we using airbnb standards for React as well? I would use the same as what we are doing there.

Comment (Ravikoduru-dcs) [reply react-native-eslint.md]: On React we are using standard library, I'll update it.

File: react-native-eslint.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Introduction
```
Comment (osm-vishnukyatannawar) [inline react-native-eslint.md:1]: Please update the main page with react native and a link to the document.

image

Check README.md file

**Review by osm-vishnukyatannawar — APPROVED** [review]

File: react-native-eslint.md (hunk 2 above)
Comment (Ravikoduru-dcs) [reply react-native-eslint.md:1]: @osm-vishnukyatannawar  Updated ReadME file. Please check when you get free time.

File: react-native-eslint.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+Once you install the ESLint extention, this plugin needs a bit further configuration. Open the VSCode settings and add the following configuration.
+
+```sh
+"eslint.validate": [
+    "javascript",
+    "javascriptreact",
+    "typescript",
+    "typescriptreact"
+]
+}
```
Comment (Abijeet) [inline react-native-eslint.md]: This formatting is off. The `}` appears to be additional

File: react-native-eslint.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  rules: {
+    // allow .js files to contain JSX code
+    "react/jsx-filename-extension": [1, { extensions: [".js", ".jsx"] }],
+  },
+};
+```
+
+# Usage
+
+Add below scripts in your package.json to run lint on all files including .js and .tsx ones and to fix them.
```
Comment (Abijeet) [inline react-native-eslint.md]: Use : `package.json`

File: react-native-eslint.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+- [@typescript-eslint/parser@latest](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/parser)
+- [eslint-plugin-react-native](https://github.com/intellicode/eslint-plugin-react-native)
+
+```sh
+npm i eslint-plugin-react@latest, @typescript-eslint/eslint-plugin@latest, eslint-config-standard@latest, eslint@^8.0.1, eslint-plugin-import@^2.25.2, eslint-plugin-n@^15.0.0, eslint-plugin-promise@^6.0.0, @typescript-eslint/parser@latest eslint-plugin-react-native
+```
+
+# Configure ESLint
+
+React Native comes with a predefined .eslintrc.js file in the root folder of your project, Open `.eslintrc.js` file and modify it like the below configuration.
```
Comment (Abijeet) [inline react-native-eslint.md]: Use `.eslintrc.js`

Please use this for all files

File: README.md [hunk 6]
```diff
@@ -1,4 +1,5 @@
 # Introduction
+
```
Comment (Abijeet) [inline README.md:2]: Put the image in the `assets` folder

File: react-native-eslint.md (hunk 5 above)
Comment (Ravikoduru-dcs) [reply react-native-eslint.md]: @Abijeet , I don't understand.. Do you want me to add .eslintrc.js file?

File: react-native-eslint.md (hunk 4 above)
Comment (Ravikoduru-dcs) [reply react-native-eslint.md]: @Abijeet , I don't understand.. Do you want me to add package.json file?

Comment (osm-vishnukyatannawar) [reply react-native-eslint.md]: @Ravikoduru-dcs  he is talking about formatting here `package.json` (inline code)

File: react-native-eslint.md (hunk 5 above)
Comment (osm-vishnukyatannawar) [reply react-native-eslint.md]: Same here.

File: react-native-eslint.md (hunk 3 above)
Comment (Ravikoduru-dcs) [reply react-native-eslint.md]: @Abijeet  comment was addressed

File: react-native-eslint.md (hunk 4 above)
Comment (Ravikoduru-dcs) [reply react-native-eslint.md]: @Abijeet  comment was addressed

File: README.md (hunk 6 above)
Comment (Ravikoduru-dcs) [reply README.md:2]: @Abijeet  comment was addressed

File: README.md [hunk 7]
```diff
@@ -8,9 +9,11 @@ This repository will document different technology specific linting configuratio
 - [React with TypeScript](react-typescript-eslint.md)
 - [PHP v8.1 (Laravel v8) - PHPCS](php-laravel-phpcs.md)
 - [Node.Js v16 (NestJs v8)](node-nest-eslint.md)
+- [React Native v0.68](react-native-eslint.md)
```
Comment (Abijeet) [inline README.md:12]: @vishnu-kyatannawar We should maintain some proper order here. Not in this PR necessarily but in general

File: react-native-eslint.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+- [@typescript-eslint/eslint-plugin@latest](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin)
+- [eslint-config-standard@latest](https://github.com/standard/eslint-config-standard)
+- [eslint-plugin-import@^2.25.2](https://github.com/import-js/eslint-plugin-import)
+- [eslint-plugin-n@^15.0.0 ](https://github.com/github/eslint-plugin-github)
+- [eslint-plugin-promise@^6.0.0 ](https://github.com/xjamundx/eslint-plugin-promise)
+- [@typescript-eslint/parser@latest](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/parser)
+- [eslint-plugin-react-native](https://github.com/intellicode/eslint-plugin-react-native)
+
+```sh
+npm i eslint-plugin-react@latest, @typescript-eslint/eslint-plugin@latest, eslint-config-standard@latest, eslint@^8.0.1, eslint-plugin-import@^2.25.2, eslint-plugin-n@^15.0.0, eslint-plugin-promise@^6.0.0, @typescript-eslint/parser@latest eslint-plugin-react-native
```
Comment (Abijeet) [inline react-native-eslint.md]: Another comment, shouldn't this be installed as `devDependencies`?

File: react-native-eslint.md (hunk 5 above)
Comment (Abijeet) [reply react-native-eslint.md]: Its fine, what I meant was use the text "`.eslintrc.js`" properly formatted throughout the document.

File: react-native-eslint.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+If the error messages are still shown up, open the source code and modify it to fix it!
+
+# VSCode plugin setup
+
+If you use VSCode, Use ESLint Extension to have the eslint information real-time in you VSCode Editor and when you save files, files are automatically fixed by ESLint rules like you execute `npm run lint:fix`
+
+<img width="502" alt="ESlint-Extension" src="assets/ESLint.png">
+
+
```
Comment (Abijeet) [inline react-native-eslint.md]: Too much spacing here. Would prefer to reduce some empty newlines

**Review by Abijeet — CHANGES_REQUESTED** [review]
Sorry for the delay, left some more comments.

File: react-native-eslint.md (hunk 5 above)
Comment (ghost) [reply react-native-eslint.md]: @Abijeet  comment was addressed

File: react-native-eslint.md (hunk 9 above)
Comment (ghost) [reply react-native-eslint.md]: @Abijeet  comment was addressed

File: react-native-eslint.md (hunk 8 above)
Comment (ghost) [reply react-native-eslint.md]: it'll be installed as devDependencies only

Comment (Abijeet) [reply react-native-eslint.md]: You are running `npm i`, that will add to dependencies right, and not `devDependencies`?

Comment (ghost) [reply react-native-eslint.md]: yes, added -D flag

Comment (Abijeet) [reply react-native-eslint.md]: I don't see that change.

Comment (ghost) [reply react-native-eslint.md]: @Abijeet  please check now

**Review by Abijeet — APPROVED** [review]
Looks good

File: README.md (hunk 7 above)
Comment (osm-vishnukyatannawar) [reply README.md:12]: Okay sure

**Pull request merged**